src/emit/ops/cfg.rs

```diff
@@ -0,0 +1,288 @@
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Result};
+use hugr::{
+    ops::{DataflowBlock, ExitBlock, OpType, CFG},
+    types::SumType,
+    HugrView, NodeIndex,
+};
+use inkwell::{basic_block::BasicBlock, values::BasicValueEnum};
+use itertools::Itertools as _;
+
+use crate::{
+    emit::{
+        func::{EmitFuncContext, RowMailBox, RowPromise},
+        EmitOp, EmitOpArgs,
+    },
+    fat::FatNode,
+};
+
+use super::emit_dataflow_parent;
+
+pub struct CfgEmitter<'c, 'd, H: HugrView> {
+    context: &'d mut EmitFuncContext<'c, H>,
+    bbs: HashMap<FatNode<'c, OpType, H>, (BasicBlock<'c>, RowMailBox<'c>)>,
+    inputs: Option<Vec<BasicValueEnum<'c>>>,
+    outputs: Option<RowPromise<'c>>,
+    node: FatNode<'c, CFG, H>,
+    entry_node: FatNode<'c, DataflowBlock, H>,
+    exit_node: FatNode<'c, ExitBlock, H>,
+}
+
+impl<'c, 'd, H: HugrView> CfgEmitter<'c, 'd, H> {
+    // Constructs a new CfgEmitter. Creates a basic block for each of
+    // the children in the llvm function. Note that this does not move the
+    // position of the builder.
+    pub fn new(
+        context: &'d mut EmitFuncContext<'c, H>,
+        args: EmitOpArgs<'c, CFG, H>,
+    ) -> Result<Self> {
+        let node = args.node();
+        let (inputs, outputs) = (Some(args.inputs), Some(args.outputs));
+
+        // create this now so that it will be the last block and we can use it
+        // to crate the other blocks immediately before it. This is just for
+        // nice block ordering.
+        let exit_block = context.new_basic_block("", None);
+        let bbs = node
+            .children()
+            .map(|child| {
+                if child.is_exit_block() {
+                    let output_row = {
+                        let out_types = node.out_value_types().map(|x| x.1).collect_vec();
+                        context.new_row_mail_box(out_types.iter(), "")?
```

Comment:
Would it work to use the `outputs` rmb here and then make ExitBlock emission a noop?

(Not necessarily suggesting to do that, just curious)

Reply:
Ah I see. I think the current solution is good, I was just wondering